gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/SpecStore.java

```diff
@@ -78,6 +78,16 @@ public interface SpecStore {
    */
   Spec updateSpec(Spec spec) throws IOException, SpecNotFoundException;
 
+  /***
+   * Update {@link Spec} in the {@link SpecStore} when current version is smaller than {@link version}.
+   * @param spec {@link Spec} to be updated.
+   * @param version largest version that current spec should be
+   * @throws IOException Exception in updating the {@link Spec}.
+   * @return Updated {@link Spec}.
+   * @throws SpecNotFoundException If {@link Spec} being updated is not present in store.
+   */
+  default Spec updateSpec(Spec spec, long version) throws IOException, SpecNotFoundException {return updateSpec(spec);};
```

Comment:
If we are able to accept request from multi host, it's possible that two user try to modify same flow at the same time, especially for patch update, this version is introduced to avoid this situation, only the first update will go through, the second one will fail with the concurrent update exception.

Reply:
For that purpose, I think version isn't the best naming for this functionality, maybe something more related to the modifiedTimestamp? e.g. modifiedWatermark? Version makes me think that it would be directly incremental.

Reason being is that there was some interest in adding versioning to the FlowSpecStore, if this is needed for users in the future for them to undo changes, then it would make sense we refer to this timestamp differently.

Reply:
So this comment is on the outdated code, so it may not be applicable. Are we trying to change our code to handle two concurrent requests? I think we can achieve this with the right mysql commands only.